### PR TITLE
refactor: remove AliasModule

### LINKED ISSUE #310: Remove AliasModule.scala
It's become unmaintainable. Instead, we should import netty directly into our code and prefix with the letter `H` incase of a conflict.

```scala
import io.netty.channel.Channel

case class HChannel(channel: Channel)
```

### PR DESCRIPTION
**Pull request by dinojohn**
fixes #310

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/http/Header.scala [hunk 1]
```diff
@@ -2,9 +2,13 @@ package zhttp.http
 
 import io.netty.buffer.Unpooled
 import io.netty.handler.codec.base64.Base64
-import io.netty.handler.codec.http.{HttpHeaderNames => JHttpHeaderNames, HttpHeaderValues => JHttpHeaderValues}
+import io.netty.handler.codec.http.{
+  DefaultHttpHeaders,
+  HttpHeaderNames => JHttpHeaderNames,
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/Header.scala]: missed this

File: zio-http/src/main/scala/zhttp/service/ChannelFactory.scala [hunk 2]
```diff
@@ -1,8 +1,10 @@
 package zhttp.service
 
-import io.netty.channel.epoll.{Epoll => JEpoll}
-import io.netty.channel.kqueue.{KQueue => JKQueue}
-import zhttp.core._
+import io.netty.channel.embedded.EmbeddedChannel
+import io.netty.channel.epoll.{Epoll => JEpoll, EpollSocketChannel}
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/service/ChannelFactory.scala]: missed this

**Review by amitksingh1490 — CHANGES_REQUESTED** [review]

**Comment by tusharmath** [comment]
Search for the following two reg-ex

```regex
import.*\{.*\n.*=>
```

```regex
import.*\{.*.*=>
```

File: example/src/main/scala/HttpsClient.scala [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+import zhttp.service.{Client, HChannelFactory, HEventLoopGroup}
 import zio._
 
 import java.io.InputStream
 import java.security.KeyStore
 import javax.net.ssl.TrustManagerFactory
 
 object HttpsClient extends App {
-  val env     = ChannelFactory.auto ++ EventLoopGroup.auto()
+  val env     = HChannelFactory.auto ++ HEventLoopGroup.auto()
```
Comment (tusharmath) [inline example/src/main/scala/HttpsClient.scala]: Public API should not have `H` prefix.

File: zio-http/src/main/scala/zhttp/service/ChannelFactory.scala [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
+import io.netty.channel.socket.nio.NioSocketChannel
+import io.netty.channel.{Channel, ChannelFactory}
 import zio.{UIO, ZLayer}
 
-object ChannelFactory {
-  def nio: ZLayer[Any, Nothing, ChannelFactory]      = Live.nio.toLayer
-  def epoll: ZLayer[Any, Nothing, ChannelFactory]    = Live.epoll.toLayer
-  def embedded: ZLayer[Any, Nothing, ChannelFactory] = Live.embedded.toLayer
-  def auto: ZLayer[Any, Nothing, ChannelFactory]     = Live.auto.toLayer
+object HChannelFactory {
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/ChannelFactory.scala]: Public API should not have `H` prefix.

File: zio-http/src/main/scala/zhttp/service/ChannelFactory.scala [hunk 5]
```diff
@@ -1,8 +1,10 @@
 package zhttp.service
 
-import io.netty.channel.epoll.{Epoll => JEpoll}
-import io.netty.channel.kqueue.{KQueue => JKQueue}
-import zhttp.core._
+import io.netty.channel.embedded.EmbeddedChannel
+import io.netty.channel.epoll.{Epoll, EpollSocketChannel}
+import io.netty.channel.kqueue.{KQueue, KQueueSocketChannel}
+import io.netty.channel.socket.nio.NioSocketChannel
+import io.netty.channel.{Channel, ChannelFactory => HChannelFactory}
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/ChannelFactory.scala]:
```suggestion
import io.netty.channel.{Channel, ChannelFactory => JChannelFactory}
```

File: zio-http/src/main/scala/zhttp/socket/WebSocketFrame.scala [hunk 6]
```diff
@@ -1,9 +1,10 @@
 package zhttp.socket
 
-import zhttp.core.{ByteBuf, _}
+import io.netty.handler.codec.http.websocketx.{WebSocketFrame => HWebSocketFrame, _}
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/socket/WebSocketFrame.scala]:
```suggestion
import io.netty.handler.codec.http.websocketx.{WebSocketFrame => JWebSocketFrame, _}
```

File: zio-http/src/main/scala/zhttp/service/server/ServerChannelFactory.scala [hunk 7]
```diff
@@ -1,8 +1,9 @@
 package zhttp.service.server
 
-import io.netty.channel.epoll.{Epoll => JEpoll}
-import io.netty.channel.kqueue.{KQueue => JKQueue}
-import zhttp.core._
+import io.netty.channel.epoll.{Epoll, EpollServerSocketChannel}
+import io.netty.channel.kqueue.{KQueue, KQueueServerSocketChannel}
+import io.netty.channel.socket.nio.NioServerSocketChannel
+import io.netty.channel.{ChannelFactory => HChannelFactory, ServerChannel}
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/server/ServerChannelFactory.scala]:
```suggestion
import io.netty.channel.{ChannelFactory => JChannelFactory, ServerChannel}
```

File: zio-http/src/main/scala/zhttp/service/package.scala [hunk 8]
```diff
@@ -1,6 +1,6 @@
 package zhttp
 
-import zhttp.core._
+import io.netty.channel.{Channel, ChannelFactory => HChannelFactory, EventLoopGroup => HEventLoopGroup, ServerChannel}
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/package.scala]:
```suggestion
import io.netty.channel.{Channel, ChannelFactory => JChannelFactory, EventLoopGroup => HEventLoopGroup, ServerChannel}
```

File: zio-http/src/main/scala/zhttp/service/Client.scala [hunk 9]
```diff
@@ -1,7 +1,8 @@
 package zhttp.service
 
-import io.netty.handler.codec.http.{HttpVersion => JHttpVersion}
-import zhttp.core._
+import io.netty.bootstrap.Bootstrap
+import io.netty.channel.{Channel, ChannelFactory => HChannelFactory, EventLoopGroup => HEventLoopGroup}
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Client.scala]:
```suggestion
import io.netty.channel.{Channel, ChannelFactory => JChannelFactory, EventLoopGroup => HEventLoopGroup}
```

File: zio-http/src/main/scala/zhttp/service/ChannelFactory.scala (hunk 5 above)
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/ChannelFactory.scala]:
```suggestion
import io.netty.channel.{Channel, ChannelFactory => JChannelFactory}
```

File: zio-http/src/main/scala/zhttp/http/Scheme.scala [hunk 10]
```diff
@@ -1,6 +1,6 @@
 package zhttp.http
 
-import io.netty.handler.codec.{http => jHttp}
+import io.netty.handler.codec.http
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Scheme.scala]:
```suggestion
import io.netty.handler.codec.http.{HttpScheme}
```

**Review by tusharmath — APPROVED** [review]

**Pull request merged**